src/com/google/javascript/jscomp/ProductionCoverageInstrumentationCallback.java

```diff
@@ -91,23 +115,99 @@ public void visit(NodeTraversal traversal, Node node, Node parent) {
       return;
     }
 
+    String functionName = functionNameStack.peek();
+
     if (node.isFunction()) {
```

Comment:
I'd find this chain of its more readable as a switch (node.getToken()), since the token names (FUNCTION, IF) pop out more.

You could put `NodeUtil.isLoopStructure(node)` in the default case since I still think using that function is a good idea.